### PR TITLE
Implement PSR-15

### LINKED ISSUE #21: Use http-interop for middlewares PSR-15
We should use the [Interop\Http\ServerMiddleware\MiddlewareInterface](https://github.com/php-fig/fig-standards/blob/master/proposed/http-middleware/middleware.md#21-psrhttpservermiddlewaremiddlewareinterface) (PSR-15).

Zend Expressive has a pretty nifty [migration guide](https://docs.zendframework.com/zend-expressive/reference/migration/to-v2/#http-interop).

### PR DESCRIPTION
**Pull request by bgaleotti**
Should close #21. Superseeds #23.

### CODE REVIEW

**Comment by sandrokeil** [comment]
Is it useful to use [Problem Details](https://framework.zend.com/blog/2017-03-23-expressive-error-handling.html) responses on errors? But then we have to implement response classes and additional dependecies.

File: composer.json [hunk 1]
```diff
@@ -41,7 +43,9 @@
     "prooph/common": "^4.0",
     "prooph/service-bus" : "^6.0",
     "psr/http-message": "^1.0",
-    "react/promise" : "^2.2"
+    "react/promise" : "^2.2",
+    "http-interop/http-middleware": "^0.4.1",
+    "fig/http-message-util": "^1.1"
   },
   "require-dev": {
     "phpunit/phpunit": "^5.7",
```
Comment (prolic) [inline composer.json]: upgrade to phpunit 6 please

Comment (prolic) [reply composer.json]: upgrade to prooph/bookdown-template please

Comment (prolic) [reply composer.json]: upgrade to prooph/common ^4.1 please

**Review by prolic — CHANGES_REQUESTED** [review]
looks good to me (besides some components that need to be upgraded), but I did not use any middleware application the last few months, so I am not 100% up to date, please as @basz for an additional review on this one.

**Review by basz — COMMENTED** [review]
Lgtm

**Review by prolic — APPROVED** [review]

**Review by codeliner — APPROVED** [review]

**Comment by prolic** [comment]
:+1:

**Comment by codeliner** [comment]
@prolic should we wait with new release till travis-ci works again? A failed master build looks not so good

**Comment by prolic** [comment]
Currently master build is red on all components, damn 7.1.3. We can wait
for 7.1.4 and wait again until travis is upgraded, too. Or we have yet
another red master, you decide


Am 10.04.2017 04:23 schrieb "Alexander Miertsch" <notifications@github.com>:

**Comment by basz** [comment]
Ik know that we can instruct travis to use a specific version of php but not how to tell is not to use a specific version as we would like

**Comment by prolic** [comment]
I tagged https://github.com/prooph/psr7-middleware/releases/tag/v0.5

**Pull request merged**